### PR TITLE
Fix searchTours

### PR DESCRIPTION
**Pull request by VNTee**
Fix names of table
Add another services for searchTours

### CODE REVIEW

File: src/main/java/com/epam/controller/ToursController.java [hunk 1]
```diff
@@ -40,7 +48,21 @@ public ModelAndView searchTours(@RequestParam String country, @RequestParam Stri
         try {
             LocalDate addStartDate = Validator.getDate(startDate, true);
             LocalDate addEndDate = Validator.getDate(endDate, true);
-            toursModel.addObject("list", toursOfferService.searchTours(null, addStartDate, addEndDate));
+            List<Hotel> myList = hotelService.getHotelsByCountry(country);
+            if(myList.size()==0&&!country.isEmpty()){
```
Comment (ghost) [inline src/main/java/com/epam/controller/ToursController.java]: It seems like business logic, please replace it to the service layer

File: src/main/java/com/epam/repository/TourOfferDAO.java [hunk 2]
```diff
@@ -3,56 +3,61 @@
...
     public int addTour(TourOffer touroffer){
-        return JdbcTemplate.update("INSERT INTO tour_offer(tour_type, start_date, end_date, price_per_unit, hotel_id, description, discount_id) " +
+        return JdbcTemplate.update("INSERT INTO "+tableName+"(tour_type, start_date, end_date, price_per_unit, hotel_id, description, discount_id) " +
                 "VALUES (?, ?, ?, ?, ?, ?, ?)",touroffer.getTourType(),touroffer.getStartDate(),touroffer.getEndDate(),
                 touroffer.getPricePerUnit(),touroffer.getHotelId(),touroffer.getDescription(),touroffer.getDiscountId());
     }
 
     public int updateTour(TourOffer touroffer){
-       return JdbcTemplate.update("UPDATE tour_ofer SET " +
+       return JdbcTemplate.update("UPDATE "+tableName+" SET " +
```
Comment (ghost) [inline src/main/java/com/epam/repository/TourOfferDAO.java:42]: really good refactoring!

File: src/main/java/com/epam/controller/ToursController.java [hunk 3]
```diff
@@ -40,7 +48,21 @@ public ModelAndView searchTours(@RequestParam String country, @RequestParam Stri
...
+                log.error("Wrong input country: "+country);
+                throw new IllegalArgumentException("Wrong input country: "+country);
+            }
```
Comment (ghost) [inline src/main/java/com/epam/controller/ToursController.java]: add unit tests for this please

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/repository/TourOfferDAO.java (hunk 2 above)
Comment (VNTee) [reply src/main/java/com/epam/repository/TourOfferDAO.java:42]: I hope this is not sarcasm

Comment (ghost) [reply src/main/java/com/epam/repository/TourOfferDAO.java:42]: nope

File: src/main/java/com/epam/controller/ToursController.java (hunk 1 above)
Comment (VNTee) [reply src/main/java/com/epam/controller/ToursController.java]: Done

File: src/main/java/com/epam/controller/ToursController.java (hunk 3 above)
Comment (VNTee) [reply src/main/java/com/epam/controller/ToursController.java]: Done

**Review by ghost — APPROVED** [review]

File: src/main/resources/columns.properties [hunk 4]
```diff
@@ -12,7 +12,7 @@ hotel.name = name
 hotel.city = city
 hotel.country = country
 hotel.numberOfStars = number_of_stars
-tourOffer.tableName = tour_offer
+tourOffer.tableName = tourOffer
```
Comment (goponenkoev) [inline src/main/resources/columns.properties:15]: hmm, what was wrong with the previous version of the name? 'Cause this one doesn't match with name convention

Comment (ghost) [reply src/main/resources/columns.properties:15]: This and other names will be fixed in another PR about fixing names, for now we need to fix the logic

**Review by ghost — APPROVED** [review]

File: src/main/resources/columns.properties (hunk 4 above)
Comment (VNTee) [reply src/main/resources/columns.properties:15]: @imbirevkotlya Do you mean that i need to make method in DAO layer more simpler?

**Review by goponenkoev — APPROVED** [review]

**Pull request closed without merging**